photutils/psf/photometry.py

```diff
@@ -428,6 +431,11 @@ def nstar(self, image, star_groups, *, mask=None):
             a `True` value indicates the corresponding element of
             ``image`` is masked.
 
+        progressbar : progressbar
+            A progressbar (optional) to show progress over the star groups.
+            Could be an `astropy.utils.console.ProgressBar` or a `tqdm`
+            progressbar.
+
```

Comment:
Until just now, I was unaware that astropy even provided a `ProgressBar`.  I think the python community has converged on `tqdm` as the progress bar standard.  I would prefer that this simply be a True/False boolean, and if True would use a `tqdm` progress bar (`tqdm` is an optional dependency).  This makes the API consistent with the `progress_bar` options already available in `photutils` (e.g. see `deblend_sources`, `SourceFinder`, `ESPFBuilder`, `ImageDepth`).  Specifically, I've used the `tqdm.auto` progress bar, which switches from a text-based progress bar for Jupyter notebooks (see, e.g. https://github.com/astropy/photutils/blob/main/photutils/utils/depths.py#L247).

Reply:
I didn't know about `tqdm.auto`; I agree that alleviates any need for user-specified progressbars in general.